packages/language-server/src/plugins/typescript/features/CompletionProvider.ts

```diff
@@ -82,77 +86,115 @@ export class CompletionsProviderImpl
         }
 
         const completionItems = completions.entries
-            .map(comp => this.toCompletionItem(comp, tsDoc.uri, position))
-            .map(comp => mapCompletionItemToParent(tsDoc, comp));
+            .map((comp) =>
+                this.toCompletionItem(fragment, comp, pathToUrl(tsDoc.filePath), position),
+            )
+            .filter(isNotNullOrUndefined)
+            .map((comp) => mapCompletionItemToParent(fragment, comp));
 
         return CompletionList.create(completionItems);
     }
 
     private toCompletionItem(
+        fragment: SnapshotFragment,
         comp: ts.CompletionEntry,
         uri: string,
-        position: Position
-    ): AppCompletionItem<CompletionEntryWithIdentifer> {
-        const { label, insertText } = this.getCompletionLableAndInsert(comp);
+        position: Position,
+    ): AppCompletionItem<CompletionEntryWithIdentifer> | null {
+        const result = this.getCompletionLableAndInsert(fragment, comp);
+        if (!result) {
+            return null;
+        }
+
+        const { label, insertText, isSvelteComp } = result;
 
         return {
             label,
             insertText,
             kind: scriptElementKindToCompletionItemKind(comp.kind),
-            sortText: comp.sortText,
             commitCharacters: getCommitCharactersForScriptElement(comp.kind),
-            preselect: comp.isRecommended,
+            // Make sure svelte component takes precedence
+            sortText: isSvelteComp ? '-1' : comp.sortText,
+            preselect: isSvelteComp ? true : comp.isRecommended,
             // pass essential data for resolving completion
             data: {
                 ...comp,
                 uri,
-                position
-            }
+                position,
+            },
         };
     }
 
-    private getCompletionLableAndInsert(comp: ts.CompletionEntry) {
-        const { kind, kindModifiers, name } = comp;
+    private getCompletionLableAndInsert(fragment: SnapshotFragment, comp: ts.CompletionEntry) {
```

Comment:
irrelevant: `getCompletionLableAndInsert` has a typo

Reply:
that is my fault lol

Reply:
Np, I will change it 😄